GStreamer decoders rank manipulation improvements

Rafostar opened the pull request:
Setting env vars from within code is never a good idea, use dedicated public methods to do so instead.
Also do not limit users to just `h264` and `vp9`. Enable all decoders that come from VA plugin.

I did small correction in code. Should be "VA plugin" (not plural). There is only one plugin, but it has multiple features in it. We are mainly interested in features that are decoders here.

I also put a small fix/workaround for `GstPlay` debug errors as part of this MR in separate commit while I was at it.

Added an experimental option to use new Nvidia stateless decoders.

jeffshee commented:
@Rafostar Thanks for the PR! LGTM

One question before merging though, 
```js
this._setPluginDecodersRank("va", Gst.Rank.PRIMARY + 3);
```
could you explain why it is needed to +3? It seems like a magic number to me.

Rafostar commented:
> could you explain why it is needed to +3? It seems like a magic number to me.

Legacy "vaapidecodebin" element has "primary + 2" rank by default. See: [https://gstreamer.freedesktop.org/documentation/vaapi/vaapidecodebin.html?gi-language=c](https://gstreamer.freedesktop.org/documentation/vaapi/vaapidecodebin.html?gi-language=c)

You need to set VA higher then that to be used (previously 300). I added comments in code explaining that in case others will be contributing to your project.

I also added an experimental option to try new Nvidia stateless decoders in latest commit (as we discussed on Clapper matrix channel). Have not tried it myself as I do not have access to Nvidia GPU right now. Please check before merging. There should be debug output printing what decoders change rank to confirm.

jeffshee commented:
Thanks! I checked with my Nvidia system and the ranks did set correctly. There was no issue in playback as well.
 
w/o stateless
```
Hanabi: changed rank: 256 -> 257 for nvvp9dec
Hanabi: changed rank: 256 -> 257 for nvvp8dec
Hanabi: changed rank: 256 -> 257 for nvh265dec
Hanabi: changed rank: 256 -> 257 for nvjpegdec
Hanabi: changed rank: 256 -> 257 for nvh264dec
Hanabi: changed rank: 256 -> 257 for nvmpeg4videodec
Hanabi: changed rank: 256 -> 257 for nvmpeg2videodec
Hanabi: changed rank: 256 -> 257 for nvmpegvideodec
```

w/ stateless
```
Hanabi: changed rank: 128 -> 257 for nvvp9sldec
Hanabi: changed rank: 128 -> 257 for nvvp8sldec
Hanabi: changed rank: 128 -> 257 for nvh265sldec
Hanabi: changed rank: 128 -> 257 for nvh264sldec
```
